src/app/auth/auth.service.ts

```diff
@@ -104,7 +111,27 @@ export class AuthService {
     // do clear user cache here
   }
 
-  private _login(body: HttpParams, serviceHeader?: string) {
+  /**
+   * @name _loginAPILogin
+   * @description Calling login API to login the user and save stacks in local storage.
+   * @param body Json Object - request parameter need to pass to login api.
+   */
+  private _loginAPILogin(body: LoginRequParams): Observable<any> {
+    body.from = 'App';
+    return this.request.post(LOGIN_API.login, body, {}, true)
+    .pipe(map(res => {
+      this.storage.set('stacks', res.stacks);
```

Comment:
the logic to save the stacks in local storage should be inside the component instead of the service.
Also, let's not use `this.storage.set()` directly, as it makes it hard to track what has been stored in the local storage, let's create a get and set function in storage.service same as others for this.